src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3697,7 +3672,7 @@ private void VisitMemberAccess(BoundExpression receiverOpt, Symbol member, bool
                 {
                     // We are supposed to track information for the node. Use whatever we managed to
                     // accumulate so far.
-                    if (resultType.IsReferenceType)
+                    if (!resultType.IsValueType)
```

Comment:
>if (!resultType.IsValueType) [](start = 20, length = 28)

It is not obvious why this change is necessary.

Reply:
Here's the [corresponding test](https://github.com/dotnet/roslyn/pull/29875/commits/a26c373c8d63af4afc00ed1804ef3d7c8dddab11#diff-4ef499b331bacc63b10e31e95a00db61): `Members_Fields_UnconstrainedType`